engine/src/main/java/io/camunda/zeebe/engine/processing/ProcessEventProcessors.java

```diff
@@ -67,7 +66,8 @@ public static TypedRecordProcessor<ProcessInstanceRecord> addProcessProcessors(
 
     final var processEngineMetrics = new ProcessEngineMetrics(zeebeState.getPartitionId());
 
-    addProcessInstanceCommandProcessor(typedRecordProcessors, zeebeState.getElementInstanceState());
+    addProcessInstanceCommandProcessor(
```

Comment:
:x: In this class (but not in any of the changed lines) `streamWriter` and `responseWriter` are no longer used. Removing them here allows to remove them from `EngineContext` as well.

Reply:
Not sure whether I can completely follow but I think what you want is done via #10000